impl/src/main/java/org/jboss/weld/bean/proxy/util/WeldDefaultProxyServices.java

```diff
@@ -183,4 +184,25 @@ public final Class<?> publicDefineClass(String name, byte[] b, int off, int len,
             }
         }
     }
+
+    /**
+     * A dummy implementation which has only one purpose - to avoid instantiating {@code DefaultClassFactory.INSTANCE}.
+     * The sole method in this class is never used as we define classes using different means that further vary
+     * between in-container (such as WildFly) and SE setups.
+     *
+     * See {@link WeldDefaultProxyServices#defineClass(Class, String, byte[], int, int)} for details on how we define
+     * classes.
+     */
+    public static class DummyClassFactoryImpl implements ClassFactory {
+
+        private DummyClassFactoryImpl() {}
+
+        // volatile because of possible multiple tread defining classes simultaneously
+        public static volatile DummyClassFactoryImpl INSTANCE = new DummyClassFactoryImpl();
```

Comment:
I am not 100% sure this needs to be `volatile` but I thought it's better safe than sorry case.
Weld can use multiple threads during bootstrap and proxy definition and there is no need to have multiple references of this dummy factory being passes around.

Reply:
Why not `final`?

Reply:
Ah, good point, that's way simpler. Done!